secboot/secboot_tpm.go

```diff
@@ -318,35 +318,107 @@ var fdeRevealKeyRuntimeMax = "2m"
 // overridden in tests
 var fdeRevealKeyCommandExtra []string
 
-// fdeRevealKeyCommand returns a *exec.Cmd that is suitable to run
-// fde-reveal-key using systemd-run
-func fdeRevealKeyCommand() *exec.Cmd {
+// runFDERevealKeyCommand returns the output of fde-reveal-key run
+// with systemd.
+//
+// Note that systemd-run in the initrd can only talk to the private
+// systemd bus so this cannot use "--pipe" or "--wait", see
+// https://github.com/snapcore/core-initrd/issues/13
+func runFDERevealKeyCommand(stdin []byte) (output []byte, err error) {
+	runDir := filepath.Join(dirs.GlobalRootDir, "/run/fde-reveal-key")
```

Comment:
I wonder if we should add something random to this too to ensure this could run in parallel (something we don't do and probably won't do but if we would chaos would endure).

Reply:
it might be a good idea, but can be a follow up given indeed that we are not going to do this in parallel right now.

Related to follow-ups, the point remains that still file should be split up, in 2 or 3 parts:  general bits, tpm bits, hook bits?